src/snowflake/connector/network.py

```diff
@@ -481,11 +481,17 @@ def request(
             HTTP_HEADER_USER_AGENT: PYTHON_CONNECTOR_USER_AGENT,
         }
         try:
-            from opentelemetry.propagate import inject
+            # SNOW-1763555: inject OpenTelemetry headers if available specifically in WC3 format
+            #  into our request headers in case tracing is enabled. This should make sure that
+            #  our requests are accounted for properly if OpenTelemetry is used by users.
+            from opentelemetry.propagate import TraceContextTextMapPropagator
 
-            inject(headers)
-        except ModuleNotFoundError as e:
-            logger.debug(f"Opentelemtry otel injection failed because of: {e}")
+            TraceContextTextMapPropagator().inject(headers)
+        except Exception:
+            logger.debug(
+                "Opentelemtry otel injection failed",
+                exc_info=True,
+            )
```

Comment:
Do we have a test about this log message? It happened in the past that we crash because of this.

Reply:
I can add one, not a problem!